Assets/Scripts/MaterialCache.cs

```diff
@@ -9,37 +11,40 @@ public class MaterialCache : MonoBehaviour
   {
     const string c_defaultTexturePath = "Textures/node/sand_01_diff_1k.jpg";
 
-    public IDictionary<int, Material> cache = new Dictionary<int, Material>();
+    public IDictionary<string, Material> cache = new Dictionary<string, Material>();
 
-    public void Load(IEnumerable<NodeClass> classes)
+    public Lazy<Material> DefaultMaterial = new Lazy<Material>(
+      () => new Material(Shader.Find("Standard")) { mainTexture = LoadDefaultTexture() });
+    
+    public void Load(IEnumerable<NodeClass> nodeClasses, IEnumerable<EdgeClass> edgeClasses)
```

Comment:
I don't like the design. This method clearly does two things

Reply:
Wdym?
It loads the textures and caches them as materials. I don;t see how there are 2 different things

Reply:
Well, it doesn't actually use both parameters at once, it just uses the first and than the next one. With simple inheritance like EdgeClass, NodeClass <: TexturizedClass or whaterever you could reduce that to one parameter method called twice. Prolly just preference though